src/adapter/src/catalog.rs

```diff
@@ -506,8 +506,15 @@ impl CatalogState {
     /// context.
     #[tracing::instrument(level = "info", skip_all)]
     pub fn parse_view_item(&self, create_sql: String) -> Result<CatalogItem, anyhow::Error> {
-        let mut session_catalog = ConnCatalog {
-            state: Cow::Borrowed(self),
+        let mut state = self.clone();
```

Comment:
Can you add a comment to this with the GH issue link for hierarchical variables, indicating that we can remove the clones once the issue is done.